src/gui/sharemanager.cpp

```diff
@@ -326,6 +326,11 @@ QDate UserGroupShare::getExpireDate() const
 
 void UserGroupShare::setExpireDate(const QDate &date)
 {
+    if (_expireDate == date) {
+        emit expireDateSet();
```

Comment:
why the `emit` of the signal is needed ?
usually in a Qt setter we would return early if the new value and old value are the sames but the notify signal would not be emitted

Reply:
@mgallien This is because I don't want to introduce too many changes. `setExpireDate` is called after the animation is triggered, and, the animation is stopped when receiving the `expireDateSet` signal (ShareUserLine::disableProgessIndicatorAnimation).  Any other ideas on how to handle that?

Reply:
is it that the annimation is triggered by the job even if the value is not changed ?
where is this code ?

Reply:
@mgallien It's in `void ShareLinkWidget::slotSetExpireDate()`